backend/src/langchain.ts

```diff
@@ -20,11 +20,12 @@ import {
 
 // store vectorised documents for each level as array
 const vectorisedDocuments = (() => {
-	let docs: DocumentsVector[] = [];
+	const docs: DocumentsVector[] = [];
 	return {
 		get: () => docs,
 		set: (newDocs: DocumentsVector[]) => {
-			docs = newDocs;
+			while (docs.length > 0) docs.pop();
+			docs.push(...newDocs);
```

Comment:
Is there a particular reason why you did this? Just curious, not upset with it.

Reply:
There was - I figured probably better to use a const within the closure; we can then be certain we are not holding onto someone else's copy of the document array, and therefore our copy will not change without us knowing about it.

With the original code, we were replacing our reference entirely, so fully at the mercy of any other code with access to that reference. Of course, the contents are only shallow-copied so the actual items could be mutated, but hopefully we are avoiding as much mutation as possible now in our codebase.

Reply:
right, makes sense